db.go

```diff
@@ -1166,7 +1179,11 @@ func (db *DB) allocate(txid common.Txid, count int) (*common.Page, error) {
 	var minsz = int((p.Id()+common.Pgid(count))+1) * db.pageSize
 	if minsz >= db.datasz {
 		if err := db.mmap(minsz); err != nil {
-			return nil, fmt.Errorf("mmap allocate error: %s", err)
+			if err == berrors.ErrMaxSizeReached {
+				return nil, err
+			} else {
+				return nil, fmt.Errorf("mmap allocate error: %s", err)
+			}
```

Comment:
```suggestion
			return nil, fmt.Errorf("mmap allocate error: %w", err)
```

Reply:
I intentionally didn't wrap ErrMaxSizeReached so that callers can catch this error specifically, otherwise they have to inspect the inside of the arbitrary error object

Reply:
I have committed this suggestion

Reply:
actually, I think it's necessary to keep this change because mmap can return ErrMaxSizedReached, then we would want that to be bubbled up un-wrapped. Does that make sense?

Reply:
another option would be to stop using %s for errors and use %w in Errorf

Reply:
> another option would be to stop using %s for errors and use %w in Errorf

This should be a generic rule of thumb. I'm OK to keep it as it's in this PR. Please feel free to change to %w in a separate PR for all such changes in a separate PR.